Library/Homebrew/tap.rb

```diff
@@ -566,31 +566,6 @@ def ruby_file?(file)
     file.extname == ".rb"
   end
 
-  # returns true if given path would present a {Formula} file in this {Tap}.
-  # accepts both absolute path and relative path (relative to this {Tap}'s path)
-  # @private
-  sig { params(file: T.any(String, Pathname)).returns(T::Boolean) }
-  def formula_file?(file)
-    file = Pathname.new(file) unless file.is_a? Pathname
-    file = file.expand_path(path)
-    return false unless ruby_file?(file)
-    return false if cask_file?(file)
-
-    file.to_s.start_with?("#{formula_dir}/")
-  end
-
-  # returns true if given path would present a {Cask} file in this {Tap}.
-  # accepts both absolute path and relative path (relative to this {Tap}'s path)
-  # @private
-  sig { params(file: T.any(String, Pathname)).returns(T::Boolean) }
-  def cask_file?(file)
-    file = Pathname.new(file) unless file.is_a? Pathname
-    file = file.expand_path(path)
-    return false unless ruby_file?(file)
-
-    file.to_s.start_with?("#{cask_dir}/")
-  end
-
```

Comment:
These were only used in the update report and `Tap#formula_files` above which has been updated.

Reply:
`cask_file?` is used for cask CI currently. There it is also used in combination with `git diff`, so may be best to move the `git diff` logic into `Tap`, so it can be used to get all changed files between two commits.

Reply:
Could you point out where specifically it's used in cask CI? I couldn't find it with a quick grep of the cask repo.

Both of these were marked `@private` so I didn't check other repos but now that you mentioned it I looked and one of them is used in the test bot repo.

```
/u/l/H/L/T/h/homebrew-test-bot (master|✔) $ git grep -F '.cask_file?'
/u/l/H/L/T/h/homebrew-test-bot (master|✔) [1]$ git grep -F '.formula_file?'
lib/tests/formulae_detect.rb:          next unless tap.formula_file?(file)
```

Reply:
I just rebased this to add back in the `.cask_file?` and `.formula_file?` methods because they are used in other repos. I made both of them more restrictive though. The use of `.formula_file?` in test bot would need to be changed first though but the current method already takes a string so that shouldn't be too hard.

The idea of combining the `git diff` logic is an interesting idea. I'd have to look to see how much logic is shared there between different use cases.